code/modules/events/vermin_migration.dm

```diff
@@ -26,3 +31,32 @@
 
 /datum/round_event/mice_migration/start()
 	SSsqueak.trigger_migration(rand(minimum_mice, maximum_mice))
+
+// Lizard filth //
+
+/datum/round_event_control/lizard_infestation
+	name = "Lizard Infestation"
+	typepath = /datum/round_event/lizard_infestation
+	weight = 10
+
+/datum/round_event/lizard_infestation
+	announceWhen = 5
+	var/lizardcount = 15
+	startWhen = 10
+
+/datum/round_event/lizard_infestation/announce()
+	priority_announce("Unusual levels of vermin have been detected near [station_name()]. Please be alert for potential contamination.", "Lifesign Alert")
+
+/datum/round_event/lizard_infestation/start()
+	var/list/turfs = list()
+	for(var/area/maintenance/A in world) // to spawn lizards in maint
+		for(var/turf/open/floor/T in A)
+			turfs += T
```

Comment:
And in the lower segment replace the var/turf/T with var/turf/open/floor/T
and do a if(!istype(T)) continue